src/Storages/Statistics/Statistics.cpp

```diff
@@ -18,6 +22,99 @@ namespace ErrorCodes
     extern const int ILLEGAL_STATISTIC;
 }
 
+enum StatisticsFileVersion : UInt16
+{
+    V0 = 0,
+};
+
+/// Version / bitmask of statistics / data of statistics /
+
+ColumnStatistics::ColumnStatistics(const StatisticsDescription & stats_desc_)
+    : stats_desc(stats_desc_), counter(0)
+{
+}
+
+void ColumnStatistics::update(const ColumnPtr & column)
+{
+    counter += column->size();
+    for (auto iter : stats)
+    {
+        iter.second->update(column);
+    }
+}
+
+Float64 ColumnStatistics::estimateLess(Float64 val) const
+{
+    if (stats.contains(TDigest))
+        return std::static_pointer_cast<TDigestStatistic>(stats.at(TDigest))->estimateLess(val);
+    return counter * ConditionEstimator::default_normal_cond_factor;
+}
+
+Float64 ColumnStatistics::estimateGreater(Float64 val) const
+{
+    return counter - estimateLess(val);
+}
+
+Float64 ColumnStatistics::estimateEqual(Float64 val) const
+{
+    if (stats.contains(Uniq) && stats.contains(TDigest))
+    {
+        auto uniq_static = std::static_pointer_cast<UniqStatistic>(stats.at(Uniq));
+        Int64 ndv = uniq_static->getNDV();
```

Comment:
ndv -> number_of_distinct_values
NDV -> NumberOfDistinctValues

Reply:
Or something else, if I didn't get it right.

Reply:
changed to `Cardinality`